src/Dialogs/Device/DeviceEditWidget.cpp

```diff
@@ -188,6 +203,7 @@ DeviceEditWidget::UpdateVisibilities() noexcept
     DeviceConfig::MaybeBluetooth(type, port_df.GetAsString());
   const bool k6bt = maybe_bluetooth && GetValueBoolean(K6Bt);
   const bool uses_speed = DeviceConfig::UsesSpeed(type) || k6bt;
+  const bool engine_sensor = (type == DeviceConfig::PortType::BLE_SENSOR) ? true : false;
```

Comment:
A rather complicated way of converting a `bool` to the same `bool`!

Reply:
Actually I would like to use `DeviceConfig::BluetoothNameStartsWith("PPGM-V2")`, because my sensor advertises it's name as `PPGM-V2` but I was not able to get it to work. If I change it to `MaybeBluetooth`  but then the k6tb option is also shown for my sensor...

Reply:
Using the name is difficult, because the name may or may not be available at the time; it's transmitted asynchronously at some random point in time *after* the device was scanned by XCSoar.

> If I change it to MaybeBluetooth but then the k6tb option is also shown for my sensor...

That description is wrong, because it sounds like the K6Bt option will *not* be shown for your sensor otherwise, but in fact it is always shown for all Bluetooth devices, no matter the condition for EngineType.

Reply:
![Screenshot_2023-05-02-12-44-39-043_org xcsoar](https://user-images.githubusercontent.com/130893503/235646598-c9223879-d999-472e-bf98-2c842042f9cf.jpg)

That's how it looks with the current implementation... No K6Bt option. Up to now there was not a single time where the BT name was not shown. Does the config not get updated on selection of BLE sensor? The BT name is stored in the config, right?

Reply:
We are talking about code in `Save()`, but that method was never called when XCSoar decided whether to show K6Bt. Therefore, it cannot possibly have an influence on that decision.
If K6Bt is invisible, them something else is wrong with your code.

> The BT name is stored in the config, right?

Is it? Tell me which DeviceConfig field (or profile key), in your opinion, stores the name of the Bluetooth device?

Reply:
This is about code in `UpdateVisibilities`. Does every BT sensor have to have the option of K6Bt?

Later on I would like to add the engine hours on this widget. If you have two or more paramotors it would be great to have engine hours of both.

Reply:
> This is about code in `UpdateVisibilities`. Does every BT sensor have to have the option of K6Bt?

Not quite - the K6Bt is a Bluetooth (classic, not BLE) device, but there's no proper way to verify (from within this widget) whether the peer is really a K6Bt, that's why XCSoar asks the user to specify that.